src/services/dashboards/dashboard-customize/modules/VariableSelectorDropdown.vue

```diff
@@ -1,114 +1,182 @@
 <template>
-    <div class="dashboard-variable-dropdown">
+    <div v-on-click-outside="hideContextMenu"
+         class="dashboard-variable-dropdown"
+         :class="{ 'open-menu': visibleMenu }"
+    >
         <button ref="targetRef"
                 class="dropdown-box"
-                :class="{ 'is-visible': state.visibleMenu, 'filled-value': state.selected.length }"
-                @click="showContextMenu"
+                :class="{ 'is-visible': visibleMenu, 'filled-value': selected.length }"
+                @click="handleChangeVisible"
         >
             <span class="variable-label">{{ variableName }}</span>
-            <span v-if="state.selected.length"
+            <span v-if="selected.length"
                   class="selected-items"
             >
-                <span class="item-for-display">{{ state.selected[0].label }}</span>
-                <p-badge v-if="state.selected.length > 1"
+                <span class="item-for-display">{{ selected[0].label }}</span>
+                <p-badge v-if="selected.length > 1"
+                         class="selected-count"
                          style-type="blue300"
                 >
-                    +{{ state.selected.length - 1 }}
+                    +{{ selected.length - 1 }}
                 </p-badge>
+                <button class="option-delete-button"
+                        @click.stop="handleClearSelected"
+                >
+                    <p-i name="ic_delete"
+                         width="1rem"
+                         height="1rem"
+                         color="inherit"
+                    />
+                </button>
             </span>
-            <span />
+
             <p-i :name="state.visibleMenu ? 'ic_arrow_top' : 'ic_arrow_bottom'"
                  :activated="state.visibleMenu"
```

Comment:
nitpick: I think these `visibleMenu` should change, too.

Reply:
Thank you !!